test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java

```diff
@@ -120,59 +120,66 @@ private static boolean isDockerEngineAvailableCheck() throws Exception {
+    }
 
-    /**
-     * Build a docker image that contains JDK under test.
-     * The jdk will be placed under the "/jdk/" folder inside the docker file system.
+     /**
+     * Build a container image that contains JDK under test.
+     * The jdk will be placed under the "/jdk/" folder inside the image/container file system.
      *
-     * @param imageName     name of the image to be created, including version tag
-     * @param dockerfile    name of the dockerfile residing in the test source;
-     *                      we check for a platform specific dockerfile as well
-     *                      and use this one in case it exists
-     * @param buildDirName  name of the docker build/staging directory, which will
-     *                      be created in the jtreg's scratch folder
+     * @param imageName         name of the image to be created, including version tag
+     * @param dockerfileContent content of the Dockerfile; use null to generate default content
      * @throws Exception
      */
-    public static void
-        buildJdkDockerImage(String imageName, String dockerfile, String buildDirName)
-            throws Exception {
+    public static void buildJdkContainerImage(String imageName, String dockerfileContent) throws Exception {
+        // image name may contain tag, hence replace ':'
+        String imageDirName = imageName.replace(":", "-");
 
-        Path buildDir = Paths.get(".", buildDirName);
+        // Create an image build/staging directory
+        Path buildDir = Paths.get(imageDirName);
         if (Files.exists(buildDir)) {
             throw new RuntimeException("The docker build directory already exists: " + buildDir);
         }
+        Files.createDirectories(buildDir);
 
-        Path jdkSrcDir = Paths.get(JDK_UNDER_TEST);
-        Path jdkDstDir = buildDir.resolve("jdk");
-
-        Files.createDirectories(jdkDstDir);
+        // Generate Dockerfile
+        if (dockerfileContent != null) {
+            Files.writeString(buildDir.resolve("Dockerfile"), dockerfileContent);
+        } else {
+            generateDockerFile(buildDir.resolve("Dockerfile"),
+                           DockerfileConfig.getBaseImageName(),
+                           DockerfileConfig.getBaseImageVersion());
+        }
 
         // Copy JDK-under-test tree to the docker build directory.
         // This step is required for building a docker image.
+        Path jdkSrcDir = Paths.get(JDK_UNDER_TEST);
+        Path jdkDstDir = buildDir.resolve("jdk");
+        Files.createDirectories(jdkDstDir);
         Files.walkFileTree(jdkSrcDir, new CopyFileVisitor(jdkSrcDir, jdkDstDir));
-        buildDockerImage(imageName, Paths.get(Utils.TEST_SRC, dockerfile), buildDir);
+
+        buildImage(imageName, buildDir);
     }
 
 
     /**
-     * Build a docker image based on given docker file and docker build directory.
+     * Build a container image based on given Dockerfile and image build directory.
      *
```

Comment:
This comment looks wrong because there is no longer a given Dockerfile.

Reply:
Thanks. I will remove this comment.